llm_studio/src/datasets/text_causal_language_modeling_ds.py

```diff
@@ -322,12 +339,18 @@ def _read_data(self, idx: int, sample: Dict) -> Dict:
             samples.insert(0, self._get_sample(int(rnd_idx)))
 
         input_ids = torch.cat([torch.cat(sample) for sample in samples])
-        prompt_mask = torch.cat(
-            [
-                torch.cat([torch.ones_like(sample[0]), torch.zeros_like(sample[1])])
-                for sample in samples
-            ]
-        ).to(torch.bool)
+
+        if self.cfg.training.use_rlhf and self.mode == "train":
```

Comment:
The `_read_data` method is a bit hard to follow. But I can also refactor after this PR has been merged.